### PR TITLE
[docs] New ML pipeline guide PR

### PR DESCRIPTION
**Pull request by odette-elementl**
## Summary & Motivation
moved changes from https://github.com/dagster-io/dagster/pull/12930

### CODE REVIEW

File: docs/content/guides/dagster/ml-pipeline.mdx [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+def training_test_data(hackernews_stories):
+    hackernews_stories = hackernews_stories
+    X = hackernews_stories.title
+    y = hackernews_stories.descendants
+    # Split the dataset to reserve 20% of records as the test set
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+    return (X_train, y_train), (X_test, y_test)
+```
+
+Next, we will take both the training and test data subsets and [tokenize the titles](https://en.wikipedia.org/wiki/Lexical_analysis) e.g. take the words and turn them into columns with the frequency of terms for each records to create [features](https://en.wikipedia.org/wiki/Feature_\(machine_learning\)) for the data. To do this, we will be using the training set to fit the tokenizer, in this case we are using [TfidfVectorizer](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html) and then transforming both the training and test set based with that tokenizer.
```
Comment (erinkcochran87) [inline docs/content/guides/dagster/ml-pipeline.mdx:106]:
```suggestion
Next, we will take both the training and test data subsets and [tokenize the titles](https://en.wikipedia.org/wiki/Lexical_analysis) e.g. take the words and turn them into columns with the frequency of terms for each record to create [features](https://en.wikipedia.org/wiki/Feature_\(machine_learning\)) for the data. To do this, we will be using the training set to fit the tokenizer. In this case, we are using [TfidfVectorizer](https://scikit-learn.org/stable/modules/generated/sklearn.feature_extraction.text.TfidfVectorizer.html) and then transforming both the training and test set based on that tokenizer.
```

A few small grammar fixes

**Review by erinkcochran87 — APPROVED** [review]
A few small grammar tweaks, but overall this looks good to me! I'll lave it to @sryza to stamp before merging. Nice job Odette!

File: docs/content/guides/dagster/ml-pipeline.mdx [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+---
+
+## Benefits of building machine learning pipelines in Dagster
+
+- Dagster makes iterating on machine learning models and testing easy, and it is designed to use during the development process.
+- Dagster has a lightweight execution model means you can access the benefits of an orchestrator, like re-executing from the middle of a pipeline and parallelizing steps while you're experimenting.
+- Dagster models data assets, not just tasks, so it understands the upstream and downstream data dependencies.
+- Dagster is a one-stop shop for both the data transformations and the models that depend on the data transformations.
+
+<Note>
```
[the remark this answers is not in the document]
Comment (odette-elementl) [reply docs/content/guides/dagster/ml-pipeline.mdx:21]: put in a note or just text?

[the remark this answers is not in the document]
Comment (odette-elementl) [reply docs/content/guides/dagster/ml-pipeline.mdx]: @erinkcochran87  Updated it - i think that looks good, thanks for the feedback

File: docs/content/_navigation.json [hunk 3]
```diff
@@ -741,10 +741,14 @@
...
           },
+          {
+          "title": "Building machine learning pipelines",
+          "path": "/guides/dagster/ml-pipeline"
+          },
           {
             "title": "Using environment variables and secrets",
             "path": "/guides/dagster/using-environment-variables-and-secrets"
-          },
+          },
```
Comment (sryza) [inline docs/content/_navigation.json:751]: I think we don't need to add this whitespace

Comment (odette-elementl) [reply docs/content/_navigation.json]: removed
